lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -2462,6 +2463,100 @@ public void detach_message_to_released_channel_is_dropped() throws AblyException
+            new ChannelWaiter(channel).waitFor(ChannelState.attached);
+            assertEquals("Verify attached state reached", channel.state, ChannelState.attached);
+
+            /* push a message to increase msgSerial */
+            channel.publish("test", "test");
+            assertEquals(1, ably.connection.connectionManager.msgSerial);
+
+            ably.close();
+            new ChannelWaiter(channel).waitFor(ChannelState.detached);
+            assertEquals(ConnectionState.closed, ably.connection.state);
+            assertEquals(1, ably.connection.connectionManager.msgSerial);
+
+            ably.connect();
+
+            new ConnectionWaiter(ably.connection).waitFor(ConnectionState.connected);
+            assertEquals(ChannelState.initialized, channel.state);
+
+            assertNull(channel.reason);
+            assertNull(ably.connection.reason);
+            assertEquals(ChannelState.initialized, channel.state);
+            assertEquals(0, ably.connection.connectionManager.msgSerial);
+        }
+    }
+
+    /*
+     * Spec: RTN11b
+     * Checks that all channels become if the state is CLOSING transitions all the channels to
+     * INITIALIZED and unsets:
+     *  - RealtimeChannel.errorReason
+     *  - Connection.errorReason
+     *  - msgSerial
+     */
+    @Test
+    public void connect_on_closing_client_should_reinitialize_channels() throws AblyException {
+        ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+        try (AblyRealtime ably = new AblyRealtime(opts)) {
+
+            /* wait until connected */
+            (new ConnectionWaiter(ably.connection)).waitFor(ConnectionState.connected);
+            assertEquals("Verify connected state reached", ably.connection.state, ConnectionState.connected);
+
+            /* create a channel and attach */
+            final Channel channel = ably.channels.get("channel");
+            channel.attach();
+            new ChannelWaiter(channel).waitFor(ChannelState.attached);
+            assertEquals("Verify attached state reached", channel.state, ChannelState.attached);
+
+            /* push a message to increase msgSerial */
+            channel.publish("test", "test");
+            assertEquals(1, ably.connection.connectionManager.msgSerial);
+
+            List<ChannelState> observedChannelStates = new ArrayList<>();
+            channel.on(stateChange -> observedChannelStates.add(stateChange.current));
+
+            List<ConnectionState> observedConnectionStates = new ArrayList<>();
+            ably.connection.on(stateChange -> observedConnectionStates.add(stateChange.current));
+
+            ably.close();
+            new ConnectionWaiter(ably.connection).waitFor(ConnectionState.closing);
+            ably.connect();
+
+            new ConnectionWaiter(ably.connection).waitFor(ConnectionState.connected);
+            assertEquals(List.of(ConnectionState.closing, ConnectionState.connecting, ConnectionState.connected), observedConnectionStates);
```

Comment:
Not sure, but it feels like the test would flake because the connection can go into `CLOSED` state, before we call 
`ably.connect` at L2545

Reply:
Okay, ran multiple times on the local, works good 👍

Reply:
I rewrote test a little, make `connect()` instantly after `close()` so there will be no chance that closed message from network comes faster than two sequential statement